### PR TITLE
feat: http instrumentation

### PR DESCRIPTION
**Pull request by mathnogueira**
This PR adds instrumentation for http operations. Right now it logs all http endpoints and all `servicer` calls. The trace is looking like that:

image

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: server/http/instrumented_controller.go [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+		func(ctx context.Context) (openapi.ImplResponse, error) {
+			return is.servicer.CreateTest(ctx, test)
+		},
+		func() []attribute.KeyValue {
+			testJson, _ := json.Marshal(test)
+			return []attribute.KeyValue{
+				attribute.String("http.request.body", string(testJson)),
+			}
+		},
+	)
```
Comment (mathnogueira) [inline server/http/instrumented_controller.go:59]: @schoren any idea if there is a way to simplify this?

File: server/http/instrumented_controller.go [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+}
```
Comment (mathnogueira) [inline server/http/instrumented_controller.go:60]: @schoren any idea if there is a way to simplify this?

File: server/http/instrumented_controller.go [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+	"context"
+	"encoding/json"
+
+	"github.com/kubeshop/tracetest/server/openapi"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type instrumentedServicer struct {
```
[the remark this answers is not in the document]
Comment (mathnogueira) [reply server/http/instrumented_controller.go:13]: I did that as a decorator for 2 reasons:

1. We ensure that no one will forget to instrument any new route that might be created. By just having the controller, you can skip instrumentation altogether. We still can have the problem of changing the names of the variables  in the controller and the instrumentation using the old names, but maybe that can be automated by using reflection;
2. To apply the single responsibility principle. Some of the functions we have in the controller are "big"

[the remark this answers is not in the document]
Comment (schoren) [reply server/http/instrumented_controller.go:13]: I agree in principle, but in practice it seems like a lot of code.

You already have [this](https://github.com/kubeshop/tracetest/pull/675/files/be504271eed40942551bb85f5fd3230a0e481e1d#diff-711e878d10773ddd9c1bb2cdfc83dbc1920b03f34f60ecad014d92bd8287d823R42), so why don't do something like:

```
func (c *customController) instrumentRoute(name, route string, f http.HandlerFunc) http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request) {
		ctx, span := c.tracer.Start(r.Context(), name)
		defer span.End()

		params := []attribute.String{}

		for k, v := range mux.Vars(r) {
			params = append(params, attribute.String(k, v))
		}

		span.SetAttributes(
			attribute.String(string(semconv.HTTPMethodKey), r.Method),
			attribute.String(string(semconv.HTTPRouteKey), route),
			attribute.String(string(semconv.HTTPTargetKey), r.URL.String()),
			params...
		)

		newRequest := r.WithContext(ctx)

		f(w, newRequest)
	}
}
```

Just a quick update to get the url params, but we could expand it to get more req/resp info in a generic way.

**Review by schoren — APPROVED** [review]

**Pull request merged**